pkg/watchdogd/watchdogd.go

```diff
@@ -217,6 +225,27 @@ func (d *Daemon) Disarm() error {
 	return (*os.Process)(d).Signal(unix.SIGUSR1)
 }
 
+// DisarmAndShutdown sends a signal to the watchdog daemon to disarm and exit
+// the process. This is synchronizing. The calling process will wait until the
+// disarm has been completed.
+func (d *Daemon) DisarmAndExit() error {
+	p := (*os.Process)(d)
+	if err := p.Signal(unix.SIGINT); err != nil {
+		return err
+	}
+	// Poll because wait syscall only works for child processes. The
+	// watchdogd process should exit almost immediately. The timeout is
+	// used for the unlikely cases such as pid reuse.
+	timeout := time.NewTimer(2 * time.Second)
+	for len(timeout.C) == 0 {
+		if _, err := os.FindProcess(p.Pid); err != nil {
```

Comment:
this is error-prone. Once a PID is gone, the OS may recycle it immediately for the next process starting up. Especially if you're waiting 2 seconds

Reply:
Really, pidfds solve this problem, but I'm not sure that's available in our kernels yet. I wonder if the watchdog could send some kind of ping back?

Reply:
Yes, it's not great. If the pid gets re-used, this would timeout and print an error message, but the behaviour would still be correct (the watchdogd process did exit).

I'm worried about abusing signals too much. In Go, I don't see a way to find out the pid of the sender.

Would it make sense to switch to an http server on localhost? I was hesitant at first because the complexity.